doc/cli_subprocess_guidelines.md

```diff
@@ -0,0 +1,110 @@
+# Azure CLI Subprocess Guidelines
+
+In certain cli modules, there are scenarios that need to call a subsystem to run commands outside cli, like getting kubectl info in aks, or deployment setup in mysql using `git` and `gh`, through python built-in [subprocess](https://docs.python.org/3/library/subprocess.html) module. Despite its simplicity and versatility, ensuring the security of applying it can be challenging in different platforms under different circumstance, and it is error-prone for developers to neglect security best practices during development.
+
+
+## Insecure Use Of Subsystem Commands Under Subprocess Module
+
+
+### Dynamic Command Construction Under Risk
+
+Assume a script that needs to read in `user_input` from users to execute a `git` related command and runs it through subprocess using shell. In regular cases, users would add a valid git command, like `git --version` or something else, but if undesired input gets logged into this system, like `--version;echo aa`, the appended `;echo aa` would be executed by the subsystem too, like below: 
+
+```commandline
+import subprocess
+user_input = input("input a git command to run: ")
+cmd = f"git {user_input}"
+subprocess.run(cmd, shell=True)
+```
+
+```commandline
+input a git command to run: --version;echo aa
+git version 2.34.1
+aa
+```
+
+This is a simple example for demonstrating the side effects in python's subsystem improper usage. And it's common for cli developers to build and execute commands dynamically from users' input in a more complicated way. When constructing and executing commands in subprocess through `shell=True`, it exposes a big security vulnerability to potential malicious users outside. 
```

Comment:
subsystem -> subprocess?

https://docs.python.org/3/library/subprocess.html doesn't mention the term "subsystem" at all.